hack/update-codegen.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```

Comment:
I realize this is the convention in other repos, but I'm not super thrilled about a new top-level directory called "hack", especially when we already have a "bin" dir. There's nothing in this script that requires it live in a "hack" directory, so my preference is to move it to "bin" and make it a bit more stylistically consistent with the rest of the scripts in that directory. Something like:

```diff
diff --git a/hack/update-codegen.sh b/bin/update-codegen.sh
similarity index 56%
rename from hack/update-codegen.sh
rename to bin/update-codegen.sh
index d6252851..5e1e6507 100755
--- a/hack/update-codegen.sh
+++ b/bin/update-codegen.sh
@@ -1,8 +1,9 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
-set -o errexit
-set -o nounset
-set -o pipefail
+set -eu
+
+bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+rootdir="$( cd $bindir/.. && pwd )"
 
 # ROOT_PACKAGE :: the package (relative to $GOPATH/src) that is the target for code generation
 ROOT_PACKAGE="github.com/linkerd/linkerd2"
@@ -11,9 +12,5 @@ CUSTOM_RESOURCE_NAME="serviceprofile"
 # CUSTOM_RESOURCE_VERSION :: the version of the resource
 CUSTOM_RESOURCE_VERSION="v1alpha1"
 
-HACK_DIR=$(dirname "${BASH_SOURCE}")
-REPO_ROOT=${HACK_DIR}/..
-
 # run the code-generator entrypoint script
-
-${REPO_ROOT}/vendor/k8s.io/code-generator/generate-groups.sh all "$ROOT_PACKAGE/controller/gen/client" "$ROOT_PACKAGE/controller/gen/apis" "$CUSTOM_RESOURCE_NAME:$CUSTOM_RESOURCE_VERSION"
+$rootdir/vendor/k8s.io/code-generator/generate-groups.sh all "$ROOT_PACKAGE/controller/gen/client" "$ROOT_PACKAGE/controller/gen/apis" "$CUSTOM_RESOURCE_NAME:$CUSTOM_RESOURCE_VERSION"
```